tests/rados/test_v1client.py

```diff
@@ -0,0 +1,103 @@
+"""
+Module to verify client connection over v1 port does not result in crashes
+"""
+
+import datetime
+import random
+import re
+import time
+
+from ceph.ceph_admin import CephAdmin
+from ceph.rados.core_workflows import RadosOrchestrator
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+    # CEPH-83594645
+    Covers:
+        - BZ-2237038
+    Test to verify client connection over v1 port does not result in crashes
+    Steps
+    1. Deploy a ceph cluster
+    2. Choose a node at random which is not already configured as a client
+    3. Copy client keyring and ceph conf file having only v1 ports
+    4. Install ceph-common package
+    5. Run ceph command using new client and monitor any crash
+    """
+    log.info(run.__doc__)
+    config = kw["config"]
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    rados_obj = RadosOrchestrator(node=cephadm)
+    log.info(
+        "Running test case to verify client connection over v1 port does not result in crashes"
+    )
+
+    try:
+        # if rhbuild.startswith("5"):
+        #     log.info("Test is not valid for Pacific, BZ yet to be fixed/raised")
+        #     return 0
+
+        # choose an osd host at random
+        osd_hosts = rados_obj.ceph_cluster.get_nodes(role="osd")
+        osd_host = random.choice(osd_hosts)
+
+        rados_obj.configure_host_as_client(host_node=osd_host)
+
+        # modify the conf file and remove references to v2
+        ceph_conf, _ = osd_host.exec_command(cmd="cat /etc/ceph/ceph.conf", sudo=True)
+        # Use regex to remove only the v2 parts within the square brackets
+        ceph_conf_v1 = re.sub(r"v2:[^,]+,?", "", ceph_conf)
+        # Remove any trailing commas left by the removal
+        ceph_conf_v1 = re.sub(r",\]", "]", ceph_conf_v1)
+        # Remove any trailing spaces
+        # ceph_conf_v1 = ceph_conf_v1.strip()
+        file_name = "/etc/ceph/ceph.conf"
+        file_ = osd_host.remote_file(sudo=True, file_name=file_name, file_mode="w")
+        file_.write(ceph_conf_v1)
+        file_.flush()
+        file_.close()
+
+        # print the content of modified ceph.conf from the chosen host
+        conf_from_node, _ = osd_host.exec_command(
+            cmd="cat /etc/ceph/ceph.conf", sudo=True
+        )
+        log.info(conf_from_node)
+
+        assert "v2" not in conf_from_node
+
+        # run a set of ceph commands
+        cmds = ["ceph status", "ceph osd tree", "ceph df"]
+        for cmd in cmds:
+            log.info(f"Running cmd: {cmd}")
+            log.info(osd_host.exec_command(cmd=cmd, sudo=True)[0])
+
+        # check for any crashes on the cluster for 120 secs
+        timeout_time = datetime.datetime.now() + datetime.timedelta(seconds=120)
+        while datetime.datetime.now() < timeout_time:
+            # list any reported crashes on the cluster, should not have any
+            crash_list = rados_obj.do_crash_ls()
+            if len(crash_list) != 0:
```

Comment:
There might be existing crashes on the cluster, which are not related to tests performed here.

We need to take into account that existing crashes present on the cluster do not account for this test failure here.

Reply:
Also, we can run the commands in the while loop. 

This way, before and after each command we can check for crashes.

Reply:
We run `crash ls-new`, so ideally we should be getting redundant crashes already present in the cluster. And the prior test crash coverage will be covered in the upcoming enhancement to cluster health check module.

Included the client command execution into the while loop